polygeist/tools/cgeist/driver.cc

```diff
@@ -332,9 +332,7 @@ static LogicalResult canonicalize(mlir::MLIRContext &Ctx,
   OptPM.addPass(polygeist::createRemoveTrivialUsePass());
   OptPM.addPass(polygeist::createMem2RegPass());
   OptPM.addPass(mlir::createCanonicalizerPass(CanonicalizerConfig, {}, {}));
-  // FIXME: This is causing infinite recursion on host code.
```

Comment:
Why is it no longer causing infinite recursion?

Reply:
Because it is not running on the host part of the code when raising is enabled. 

`OptPM` in that case is only running on the GPU module: 
`(SYCLRaiseHost) ? PM.nest<gpu::GPUModuleOp>().nestAny() : PM.nestAny();`